devolutions-gateway/src/api/network_scan.rs

```diff
@@ -0,0 +1,123 @@
+use std::net::IpAddr;
+
+use axum::{
+    extract::{ws::Message, State, WebSocketUpgrade},
+    response::Response,
+};
+
+use network_scanner::scanner::NetworkScannerParams;
+use serde::Serialize;
+
+use crate::{http::HttpError, DgwState};
+
+pub async fn handler(
+    _: State<DgwState>,
+    ws: WebSocketUpgrade,
+    query_params: axum::extract::Query<NetworkScanQueryParams>,
+) -> Result<Response, HttpError> {
```

Comment:
issue: Seems like we are not checking for the NETSCAN token. See the `get_sessions` route for an example using an extractor.

Reply:
updated